### PR TITLE
[docs] Create a new app: Add improvements to the docs page

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Fixes ENG-5384

# How

This PR 
- refactors and improves the docs page to use a more clear and concise language
- it fixes some unclear statements around how to debug changes not being shown on the device section
- converts internal links to use full paths
- remove emojis

# Test Plan

The changes have been tested by running docs locally.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/get-started/create-a-new-app.md [hunk 1]
```diff
@@ -19,56 +23,60 @@ At this point we should have Expo CLI installed on our development machine and t
 
 ## Starting the development server
 
+Start the development server by running the following command:
+
 <Terminal cmd={['$ expo start']} />
 
-When you run `expo start` (or `npm start`), Expo CLI starts Metro Bundler, which is an HTTP server that compiles the JavaScript code of our app using [Babel](https://babeljs.io/) and serves it to the Expo app.
+When you run `expo start` (or `npm start`), Expo CLI starts [Metro Bundler](/guides/how-expo-works/#metro-bundler). This bundler is an HTTP server that compiles the JavaScript code of your app using [Babel](https://babeljs.io/) and serves it to the Expo app. More information on how [Expo Development server](/guides/how-expo-works/#expo-development-server) works is available.
```
[the remark this answers is not in the document]
Comment (Simek) [reply docs/pages/get-started/create-a-new-app.md:30]: Also this sentence feels like it has been cut. Maybe we can add "here" at the end and move link to this word, or maybe we can rephrase it to more CTA form, for example: "To learn more about how EDS works check out [this guide]."

[the remark this answers is not in the document]
Comment (kbrandwijk) [reply docs/pages/get-started/create-a-new-app.md:30]: Links on 'here' are against (one of our) style guide(s). But I agree that the sentence feels cut off.

[the remark this answers is not in the document]
Comment (amandeepmittal) [reply docs/pages/get-started/create-a-new-app.md]: Great points! By following our style guide and the CTA, I think a phrasing that could work here could be "Learn more about how [Expo Development Server](/guides/how-expo-works/#expo-development-server) works."

I made this change in the PR, LMK if this is good or not.

**Review by Simek — APPROVED** [review]
LGTM, thank you for the updates! 👍
